sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -360,56 +361,55 @@ def _get_snapshots(
         Returns:
             A dictionary of snapshot ids to snapshots for ones that could be found.
         """
-        query = (
-            exp.select(exp.column("snapshot", table="snapshots"))
-            .from_(exp.to_table(self.snapshots_table).as_("snapshots"))
-            .where(
-                None
-                if snapshot_ids is None
-                else self._snapshot_id_filter(snapshot_ids, "snapshots")
+        for where in (
```

Comment:
Doesn't this only return snapshots from the last batch since we do:
```
snapshots: t.Dict[SnapshotId, Snapshot] = {}
```
on each iteration?